templates/components/form/form.jsx

```diff
@@ -25,30 +27,47 @@ var Form = React.createClass({
     return {
       username: null,
       password: null,
-      email: null
+      email: null,
+      checked: false,
+      dirty: {}
     };
   },
   buildFormElement: function(key, i) {
-    var id = Object.keys(this.props.fields[i]);
+    // we always expect this.props.fields[i] to be one object with one property.
+    var self = this;
+    var id = Object.keys(this.props.fields[i])[0];
     var placeholder = this.props.fields[i][id].placeholder;
     var type = this.props.fields[i][id].type;
     var label = this.props.fields[i][id].label;
     var labelPosition = this.props.fields[i][id].labelPosition;
-    var className = this.getIconClass(id);
+    var errorMessage = this.props.fields[i][id].errorMessage;
+    var isDisabled = this.props.fields[i][id].disabled;
+    var focus = this.props.fields[i][id].focus;
+    var username = this.props.fields[i][id].username;
+    this.passChecked = this.props.fields[i][id].checked;
 
-    function foo() {
-      console.log('foo');
-      this.handleValidation(id);
+    function dirty(err, valid) {
```

Comment:
this should not be a function that is redeclared every time `render` is called, it needs to be moved out (and then we can ditch `self`, too, it'll be autobound to the component)